### PR TITLE
Add support for bridge jobs

### PR DESCRIPTION
**Pull request by kagouball**
Add support for bridge jobs
Pipeline bridges doc : https://docs.gitlab.com/ee/api/jobs.html#list-pipeline-bridges

### CODE REVIEW

**Comment by kagouball** [comment]
Mocks are coming

File: NGitLab/Models/Job.cs [hunk 1]
```diff
@@ -1,47 +1,14 @@
...
-        [DataMember(Name = "stage")]
-        [JsonPropertyName("stage")]
-        public string Stage;
-
-        [DataMember(Name = "coverage")]
-        [JsonPropertyName("coverage")]
-        public double? Coverage;
+        [DataMember(Name = "pipeline")]
+        [JsonPropertyName("pipeline")]
+        public JobPipeline Pipeline;
```
Comment (Toa741) [inline NGitLab/Models/Job.cs]: `pipeline` property seems also available on bridges

File: NGitLab/Models/Bridge.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+
+namespace NGitLab.Models
+{
+    [DataContract]
+    public class Bridge : JobCommon
+
+    {
+        [DataMember(Name = "downstream_pipeline")] [JsonPropertyName("downstream_pipeline")]
```
Comment (Toa741) [inline NGitLab/Models/Bridge.cs]:
```suggestion
        [DataMember(Name = "downstream_pipeline")]
        [JsonPropertyName("downstream_pipeline")]
```

File: NGitLab/IPipelineClient.cs [hunk 3]
```diff
@@ -102,5 +102,18 @@ public interface IPipelineClient
         /// </summary>
         /// <param name="pipelineId">ID of the pipeline</param>
         TestReport GetTestReports(int pipelineId);
+
+        /// <summary>
+        /// Returns the bridges of a pipeline.
+        /// </summary>
+        /// <param name="query"></param>
+        IEnumerable<Bridge> GetBridges(PipelineJobQuery query);
```
Comment (meziantou) [inline NGitLab/IPipelineClient.cs]: This method can be removed. `GetBridgesAsync` can be enumerated synchronously. Indeed, `GitLabCollectionResponse` implements both `IEnumerable` and `IAsyncEnumerable`.

File: NGitLab/Models/JobCommon.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Runtime.Serialization;
+using System.Text.Json.Serialization;
+
+namespace NGitLab.Models
+{
+    [DataContract]
+    public class JobCommon
```
Comment (meziantou) [inline NGitLab/Models/JobCommon.cs]: We should match the GitLab naming (https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/api/entities/ci/job_basic.rb)

```suggestion
    public class JobBasic
```

Comment (meziantou) [inline NGitLab/Models/JobCommon.cs]: Adding a base entity and moving fields and nested class is a binary breaking change. 

@jairbubbles Do you think this is a problem? If not, should we replace the fields by properties?

Comment (jairbubbles) [reply NGitLab/Models/JobCommon.cs]: I don't think it's too problematic, the jobs API is probably not used in many places. Yes, let's switch to properties in the meantime.

File: NGitLab/IPipelineClient.cs (hunk 3 above)
Comment (jairbubbles) [reply NGitLab/IPipelineClient.cs]: Indeed, I forgot about it and the "Async" was misleading. We should rename it to "AsyncOrNot" 😛

**Comment by kagouball** [comment]
Mocks  in next PR

File: NGitLab/Impl/PipelineClient.cs [hunk 5]
```diff
@@ -189,5 +189,25 @@ public TestReport GetTestReports(int pipelineId)
...
+        {
+            var url = CreateGetBridgesUrl(query);
+            return _api.Get().GetAllAsync<Bridge>(url);
+        }
+
+        private string CreateGetBridgesUrl(PipelineJobQuery query)
+        {
+            var url = $"{_pipelinesPath}/{query.PipelineId}/bridges";
+            url = Utils.AddParameter(url, "scope", query.Scope);
+            url = Utils.AddParameter(url, "include_retried", query.IncludeRetried);
```
Comment (Toa741) [inline NGitLab/Impl/PipelineClient.cs]: From the doc, `include_retried` seems not available for bridges query :/

Comment (jairbubbles) [reply NGitLab/Impl/PipelineClient.cs]: Let's make a `PipelineBridgeQuery` then!

**Pull request merged**